client/rest/src/main/java/org/elasticsearch/client/Node.java

```diff
@@ -184,19 +184,70 @@ public Roles(final Set<String> roles) {
         }
 
         /**
-         * Teturns whether or not the node <strong>could</strong> be elected master.
+         * Returns whether or not the node <strong>could</strong> be elected master.
          */
         public boolean isMasterEligible() {
             return roles.contains("master");
         }
         /**
-         * Teturns whether or not the node stores data.
+         * Returns whether or not the node stores data.
+         * @deprecated use {@link #hasDataRole()} or {@link #containsData()}
          */
+        @Deprecated
         public boolean isData() {
             return roles.contains("data");
         }
+
+        /**
+         * @return true if node has the "data" role
+         */
+        public boolean hasDataRole() {
+            return roles.contains("data");
+        }
+
+        /**
+         * @return true if node has the "data_content" role
+         */
+        public boolean hasDataContentRole() {
+            return roles.contains("data_content");
+        }
+
+        /**
+         * @return true if node has the "data_hot" role
+         */
+        public boolean hasDataHotRole() {
+            return roles.contains("data_hot");
+        }
+
+        /**
+         * @return true if node has the "data_warm" role
+         */
+        public boolean hasDataWarmRole() {
+            return roles.contains("data_warm");
+        }
+
+        /**
+         * @return true if node has the "data_cold" role
+         */
+        public boolean hasDataColdRole() {
+            return roles.contains("data_cold");
+        }
+
+        /**
+         * @return true if node has the "data_frozen" role
+         */
+        public boolean hasDataFrozenRole() {
+            return roles.contains("data_frozen");
+        }
+
+        /**
+         * @return true if node stores any type of data
+         */
+        public boolean containsData() {
```

Comment:
Super pedantic, but maybe we can call this `canContainData`, since it could be confused as "actually having data on disk"

Reply:
thanks for the additional review. done on [`2cf6ef8` (#66947)](https://github.com/elastic/elasticsearch/pull/66947/commits/2cf6ef838cb5f501d35731054ac9747b93f657e8)